ocs/agents/aggregator/drivers.py

```diff
@@ -335,9 +336,16 @@ def save_to_block(self, data):
             These must match -- in this instance both the word 'test'.
 
         Args:
-            data (dict): data dictionary from incoming data queue
-
+            data (dict):
+                data dictionary from incoming data queue
+            message_idx (int, optional):
+                message index for the incoming data. If a message index is repeated, the data will be ignored.
         """
+        if message_idx is not None:
+            if message_idx == self.last_message_idx:  # Ignore duplicate messages
```

Comment:
Consider `<=` here?

That could lead to out-of-order things getting lost.  But if things are out of order, this duplicate protection won't work anyway.

I don't think things can arrive out of order though.